### PR TITLE
Fix advertising slave connection interval in Nordic

### PR DESCRIPTION
**Pull request by ravibhagavandas**
Fix advertising slave connection interval in Nordic

Description
-----------
Nordic port was not including slave connection interval range in advertisement.
PR fixes this.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c [hunk 1]
```diff
@@ -850,8 +850,26 @@ ret_code_t prvBTAdvDataConvert( ble_advdata_t * xAdvData,
     {
         /* The first two UUID go to the initial advertisement packet */
         xAdvData->uuids_complete = xCompleteUUIDS;
-        xAdvData->p_slave_conn_int->max_conn_interval = pxParams->ulMaxInterval;
-        xAdvData->p_slave_conn_int->min_conn_interval = pxParams->ulMinInterval;
+
+        if( ( pxParams->ulMaxInterval != 0 ) && ( pxParams->ulMinInterval != 0 ) )
+        {
```
Comment (alfred2g) [inline vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c]: if both values (ulMaxInterval and ulMinInterval) are 0's, both the if and the else will not be executed.
(not sure if it is a possibility)

Comment (ravibhagavandas) [reply vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c]: if both values are zeros, it means the connection interval won't be included in advertisement, so its a no-action.

Comment (yuhui-zheng) [reply vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c]: I'd have an else branch and call it out it is intended to do nothing...

Comment (ravibhagavandas) [reply vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c]: Good callout. Added an empty else.

**Pull request merged**